src/main/java/com/team766/controllers/ExtendableCanSparkMaxSmartMotionPIDController.java

```diff
@@ -0,0 +1,283 @@
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void setFF(final double ff) {
+		pid.setFF(ff);
+	}
+
+	/*
+	 * Method to update the value of the deadzone for the PIDController
+	 * This deadzone is in hall encoder units.
+	 * @param dz the value of the deadzone in hall encoder units
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void setDeadzone(final double dz) {
+		deadzone = dz;
+	}
+
+	/*
+	 * This is a method to update the minimum and maximum output ranges of motorpower for the mechanism
+	 * @param min the minimum value for motor power (make sure that it is negative if you want the mechanism to be able to go both directions)
+	 * @param max the maximum value for motor power (make sure that it is positive if you want the mechanism to be able to go both directions)
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void setOutputRange(final double min, final double max) {
+		pid.setOutputRange(min, max);
+	}
+
+	/*
+	 * This is the method to set the minimum and maximum locations of the mechanism.
+	 * Please check your subclasses documentation for what units this is supposed to be in (usually whichever one is used as an input for the setSetpoint method)
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void setMinMaxLocation(final double min, final double max) {
+		maxPos = max;
+		minPos = min;
+	}
+
+	/*
+	* This method is used to set the maximum velocity of the mechanism for use with PID
+	* @param max the value of the maximum amount of velocity for the motorcontroller
+	* The method also sets the minimum velocity to zero, as a failsafe.
+	* @author Max Spier - 9/9/2023
+	*/
+	public void setMaxVelocity(final double max) {
+		pid.setSmartMotionMaxVelocity(max, 0);
+		pid.setSmartMotionMinOutputVelocity(0, 0);
+	}
+
+	/*
+	 * This is the method to set the maximum acelleration of the robot for the PIDController
+	 * @param max the value for the maximum acelleration
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void setMaxAccel(final double max) {
+		pid.setSmartMotionMaxAccel(max, 0);
+	}
+
+	/*
+	 * This is the method where you update the setpoint position from the subclass
+	 * @param x the setpoint
+	 * @author Max Spier - 9/9/2023
+	 */
+	protected void updateSetpointFromSubclass(double x){
+		x = MathUtil.Clamp(x, minPos, maxPos);
+		setPointPosition = x;
+	}
+
+	/*
+	 * Failsafe for the PIDController to stop it.
+	 * @author Max Spier - 9/9/2023
+	 */
+	public void stop() {
+		theState = PIDSTATE.OFF;
+	}
+
+	/*
+	 * This method does the antigrav
+	 * @author Max Spier - 9/9/2023
+	 */
+	private void antigrav(){
+		mc.set(curAntigrav);
+	}
+
+	/*
+	 * This method sets the antigrav power amount.
+	 * It gets the value for this from the subclass
+	 * @param passedValueFromSubclass the power to set the motorcontroller for antigrav
+	 * @author Max Spier - 9/9/2023
+	 */
+	protected void updateAntigrav(double passedValueFromSubclass){
+		curAntigrav = passedValueFromSubclass
+	}
+
+	/*
```

Comment:
these comment format is good / standard and should be applied to all public and protected methods

Reply:
yes